### PR TITLE
Modal Page with transparent background

### LINKED ISSUE #7878: [Bug] Page not popped on iOS 13 FormSheet swipe down
### Description

On iOS 13, when the user swipes down on a modal page with a FormSheet [presentation style](https://docs.microsoft.com/en-us/xamarin/xamarin-forms/platform/ios/ipad-page-presentation-style) to dismiss it, the NavigationPage does not raise a Popped event. Instead, the page is gone, but the NavigationPage's state acts as if the page is still being displayed.

### Steps to Reproduce

1. Push a FormSheet style modal page, for example, using an extension method like this:

``` c#
public static NavigationPage PushModal(this Page source, Page modalRoot)
{
	var navigationPage = new NavigationPage(modalRoot);
	Xamarin.Forms.PlatformConfiguration.iOSSpecific.Page.SetModalPresentationStyle(navigationPage.On<Xamarin.Forms.PlatformConfiguration.iOS>(), Xamarin.Forms.PlatformConfiguration.iOSSpecific.UIModalPresentationStyle.FormSheet);
	source.Navigation.PushModalAsync(navigationPage);
	return navigationPage;
}
```

2. Run the app. Don't close the page by pressing a button or other action that causes `NavigationController.PopModalAsync` to be called. Instead, swipe down on the page.

### Expected Behavior
The page is popped from the navigation stack, and `NavigationController.Popped` is raised.

### Actual Behavior
The page remains on the navigation stack, and `NavigationController.Popped` is not raised.

### Basic Information

- Version with issue: XF 4.3.0.851321-pre3
- Last known good version: none
- IDE: VS 16.3.2
- Platform Target Frameworks: 
  - iOS:  13.0
- Affected Devices: iPhone X

### LINKED ISSUE #9307: [Bug] `OnAppearing` does not fire after a `UIModalPresentationStyle.FormSheet` Page is Dismissed
### Description

On iOS, When a Modal Page using `UIModalPresentationStyle.FormSheet` is dismissed, `OnAppearing` does not fire on the previous page.

| Navigation Mode | `UIModalPresentationStyle` | Behavior After Page is Dismissed |
| ----------- | ---------------------------- | -------------------- |
| `Navigation.PushAsync` | N/A | ✅ `OnAppearing` Fires on Previous Page |
| `Navigation.PushModalAsync` | `UIModalPresentationStyle.FullScreen` | ✅ `OnAppearing` Fires on Previous Page |
| `Navigation.PushModalAsync` | `UIModalPresentationStyle.FormSheet` | 💥 `OnAppearing` Does Not Fire on Previous Page |

### Steps to Reproduce

1. Download/Clone this repo: https://github.com/brminnick/UIModalPresentationStyleFormSheetOnAppearingRepro
2. In Visual Studio, open `OnAppearingRepro.sln`
3. Build/deploy `OnAppearingRepro.iOS` to an iOS Simulator or Device
4. On the iOS app, confirm the following alert appears: `OnAppearing Fired`
5. On the iOS app, dismiss the alert
6. On the iOS app, click the `Push Page` Button to push a new `ContentPage` via `Navigation.PushAsync`
7. On the iOS app, click the `Back` button
8. On the iOS app, confirm the following alert appears: `OnAppearing Fired`
9. On the iOS app, dismiss the alert
10. On the iOS app, click the `Push Modal Full Screen Sheet Page` Button to push a new `ContentPage` using `UIModalPresentationStyle.FullScreen` via `Navigation.PushModalAsync`
11. On the iOS app, click the `Back` button
12. On the iOS app, confirm the following alert appears: `OnAppearing Fired`
13. On the iOS app, dismiss the alert
14. On the iOS app, click the `Push Modal Form Sheet Page` Button to push a new `ContentPage` using `UIModalPresentationStyle.FormSheet` via `Navigation.PushModalAsync`
15. On the iOS app, click the `Back` button
16. On the iOS app, confirm the following alert *does not* appear: `OnAppearing Fired`

### Expected Behavior

When a `UIModalPresentationStyle.FormSheet` page is dismissed, `OnAppearing` should fire on the previous page to match the behavior of `UIModalPresentationStyle.FullScreen`.

### Actual Behavior

When a `UIModalPresentationStyle.FormSheet` page is dismissed, `OnAppearing` does not fire on the previous page.

### Basic Information

- Version with issue: v4.4.0.991537
- Last known good version: N/A
- Platform Target Frameworks: 
  - iOS:  11.3.1

### Reproduction Link

https://github.com/brminnick/UIModalPresentationStyleFormSheetOnAppearingRepro/tree/master

### PR DESCRIPTION
**Pull request by jfversluis**
### Description of Change ###

Takeover from #6558

### Issues Resolved ### 

- closes #6558
- fixes #7878 
- fixes #9307

### PR Checklist ###

- [x] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

**Comment by jfversluis** [comment]
Added DNM because I need to process a last piece of feedback from the review on the old PR by @PureWeen, here: https://github.com/xamarin/Xamarin.Forms/pull/6558#discussion_r347389075

Other than that I think I have processed all feedback

**Comment by YZahringer** [comment]
UWP support is also planned?

**Comment by jsuarezruiz** [comment]
@YZahringer Yes.
transparent-modal-uwp

**Comment by sasivishnu** [comment]
@jfversluis  @PureWeen any update on this? we are on track of getting this in first 4.5 release? Thanks..

**Comment by PureWeen** [comment]
As of now this is planned for 4.6

@jsuarezruiz it looks like we have some UI tests failures. Can you rebase against master and we can rerun?

**Comment by jfversluis** [comment]
These tests fail for iOS 11 and 12 and I also have seen these fail in the previous test run. Oddly enough for iOS 13 there are others and they seem unrelated. I will look into them though.

image

**Comment by jfversluis** [comment]
OK, my local UITests are giving me problems so can't really look into this right now. One thing I did find on the failing iOS13 test for issue 1648 is that the `OnAppearing` isn't fired anymore with the new modal look. Also backed by: https://fluffy.es/4-ios13-breaking-changes/

> And after the dismissal, the viewDidAppear function will not be called on the presenting view controller (the view controller that presents the modal), if you are doing UI changes in this function, those changes might not be implemented!

**Comment by jfversluis** [comment]
Merged this with changed from #7923 and I think it all plays nicely together now. Appearing and Disappearing some to work correctly on iOS, hopefully the tests will reflect that 🤞

**Comment by dansiegel** [comment]
great to see some traction on this! really looking forward to seeing this merged!

**Comment by PureWeen** [comment]
Android tests passed here
https://dev.azure.com/xamarin/public/_releaseProgress?_a=release-pipeline-progress&releaseId=1392

Which is why most recent run only has iOS

File: Xamarin.Forms.Platform.iOS/Platform.cs [hunk 1]
```diff
@@ -131,7 +131,7 @@ async Task<Page> INavigation.PopModalAsync(bool animated)
 				Page.GetCurrentPage()?.SendAppearing();
 
 			return modal;
-		}
+		}
```
Comment (PureWeen) [inline Xamarin.Forms.Platform.iOS/Platform.cs]:
```suggestion
		}
```

**Comment by mhrastegary77** [comment]
> @YZahringer Yes.
> transparent-modal-uwp

The current UWP Modal pages doesn't have a back button(plus Android , iOS) or a page animation, but it does exist in your GIF! Is it planned too?

**Comment by dinesh4official** [comment]
@PureWeen , @samhouts  can you please share any documentation regarding this one?

**Comment by dinesh4official** [comment]
@PureWeen , @samhouts , Can you check the below report?

https://github.com/xamarin/Xamarin.Forms/issues/10861

**Comment by gentledepp** [comment]
@jfversluis this introduces a bug when you mave more than one modal level....
I will create a PR soon - please have an additional look!

**Pull request merged**